### DISCUSSION TITLE
Sharepoint Files  (from Microsoft Teams)

### DISCUSSION THREAD

**Discussion by tdoniger** [abraunegg/onedrive#2681] [Q&A] [answered]
Is it possible to use the onedrive client to backup files from UNIX to a sharepoint file system (from teams) not onedrive?
I am able to login using the client but unable to synchronize:
ERROR: Unable to query OneDrive to initialize application

**Comment by abraunegg** [comment, marked as answer]
@tdoniger
> Is it possible to use the onedrive client to backup files from UNIX to a sharepoint file system (from teams) not onedrive? 

Firstly, to use *any* SharePoint site, the client has to be configured with the SharePoint details - without this, accessing SharePoint is not possible. 

This being said, this is a Microsoft OneDrive client, *not* a Microsoft Teams client. Just because OneDrive and Microsoft Teams can utilise SharePoint, it does not mean that this client can necessarily access Microsoft Teams SharePoint Sites as they are generally only ever exposed via Microsoft Teams. Whilst both technologies utilise the Microsoft Graph API, the API access permissions that this client gets authenticated with are *highly* specific to Microsoft OneDrive / SharePoint and *not* Microsoft Teams. You may be in luck that it is a more generic SharePoint site that is being shared between systems (OneDrive + Teams), thus the API access permissions this client gets authorised with might give you access to that SharePoint site.

> I am able to login using the client but unable to synchronize:
> ERROR: Unable to query OneDrive to initialize application

This message specifically comes from being unable to query your OneDrive Account to obtain your account ID and your root drive ID that is associated with your account. The client generates this error when this data is not available, thus the client cannot initialise. At this point it has nothing to do with even looking at any API data for SharePoint (the client cannot query the OneDrive API to even try).

You will need to go back to your organisation and ask to have OneDrive enabled, then configure the client to access your OneDrive account, then follow the specific steps to configure SharePoint access. If your desired SharePoint site is visible, you will then be able to configure the client to access the SharePoint site you are trying to access to achieve the outcome you are looking for.

If the SharePoint site is not visible, then you are out of luck and are potentially better off attempting to use the Microsoft Teams client for your platform it it exists.
